pkg/cmd/plan.go

```diff
@@ -0,0 +1,212 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/ipfs/testground/pkg/api"
+	"github.com/ipfs/testground/pkg/config"
+
+	"github.com/BurntSushi/toml"
+	"github.com/go-git/go-git/v5"
+	gitcfg "github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/mattn/go-zglob"
+	"github.com/urfave/cli/v2"
+)
+
+var PlanCommand = cli.Command{
+	Name:  "plan",
+	Usage: "plan management",
+	Subcommands: cli.Commands{
+		&cli.Command{
+			Name:  "create",
+			Usage: "create plan `PLAN_NAME`",
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:     "module",
+					Usage:    "Create module named `MODULE_NAME`",
+					Required: false,
+					Value:    "github.com/your/module/name",
+				},
+			},
+			Action: createCommand,
+		},
+		&cli.Command{
+			Name:   "import",
+			Usage:  "`GIT_REPO` [local_repo]",
+			Action: importCommand,
+		},
+		&cli.Command{
+			Name:   "branch",
+			Usage:  "`LOCAL_REPO` `BRANCH_NAME`",
+			Action: importCommand,
+		},
+		&cli.Command{
+			Name:   "rm",
+			Usage:  "`LOCAL_REPO`",
+			Action: rmCommand,
+		},
+		&cli.Command{
+			Name:   "list",
+			Usage:  "enumerate all test cases known to the client",
+			Action: listCommand,
+		},
+	},
+}
+
+func createCommand(c *cli.Context) error {
+	if c.Args().Len() != 1 {
+		return errors.New("missing reuired argument PLAN_NAME")
+	}
+
+	cfg := &config.EnvConfig{}
+	if err := cfg.Load(); err != nil {
+		return err
+	}
+
+	pdir := filepath.Join(cfg.Dirs().Plans(), c.Args().First())
+	_, err := git.PlainInit(pdir, false)
+	if err != nil {
+		return err
+	}
+
+	fmap := map[string]string{
+		"manifest.toml": TEMPLATE_MANIFEST_TOML,
+		"main.go":       TEMPLATE_MAIN_GO,
+		"go.mod":        TEMPLATE_GO_MOD,
+	}
+
+	for fn, ts := range fmap {
+		tmpl, err := template.New(fn).Parse(ts)
+		if err != nil {
+			return err
+		}
+		f, err := os.Create(filepath.Join(pdir, tmpl.Name()))
+		if err != nil {
+			return err
+		}
+		tmpl.Execute(f, c.String("module"))
+		f.Close()
+	}
+	return nil
+}
+
+func importCommand(c *cli.Context) error {
+	if c.Args().Len() < 1 {
+		return errors.New("missing reuired argument GIT_REPO")
+	}
+	gitURL := c.Args().First()
+
+	var gitDir string
+	if c.Args().Len() > 1 {
+		gitDir = c.Args().Get(1)
+	} else {
+		sl := strings.Split(gitURL, "/")
+		gitDir = sl[len(sl)-1]
+	}
+
+	cfg := &config.EnvConfig{}
+	if err := cfg.Load(); err != nil {
+		return err
+	}
+
+	cloneOpts := git.CloneOptions{
+		URL: c.Args().First(),
+	}
+
+	_, err := git.PlainClone(filepath.Join(cfg.Dirs().Plans(), gitDir), false, &cloneOpts)
+	if err != nil {
+		return err
+	}
```

Comment:
We probably want to check if the target already exists, and fail if so.

Reply:
Unless the git clone already does that (which now that I think of it, it probably does), but we may not want to leave it up to Git. I'd rather make that check explicit.

Reply:
Git already does this, and it reports the message `repository already exists`

Reply:
👍

Reply:
I'm marking this as resolved, then.